pom.xml

```diff
@@ -6,96 +6,83 @@
     <artifactId>abacatepay-java-sdk</artifactId>
-    <version>1.0-SNAPSHOT</version>
-    <packaging>jar</packaging>
+    <version>3.0-SNAPSHOT</version>
+    <packaging>jar</packaging> <!-- Garantindo que o JAR não seja executável -->
 
     <name>abacatepay-java-sdk</name>
     <description>AbacatePay Java SDK</description>
-    <licenses>
-        <license>
-            <name>The MIT License</name>
-            <url>https://opensource.org/licenses/MIT</url>
-        </license>
-    </licenses>
-    <developers>
-        <developer>
-            <id>joel_filho</id>
-            <name>Joel Filho</name>
-            <email>joelfilho02812@gmail.com</email>
-        </developer>
-    </developers>
 
-    <properties>
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.target>11</maven.compiler.target>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    </properties>
+    <repositories>
+        <repository>
+            <id>jitpack.io</id>
+            <url>https://jitpack.io</url>
+        </repository>
+    </repositories>
 
     <dependencies>
+        <!-- Dependências principais -->
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+            <version>4.9.3</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>20240303</version>
+        </dependency>
+
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
             <version>2.13.4.2</version>
         </dependency>
-        <dependency>
-            <groupId>com.fasterxml.jackson.datatype</groupId>
-            <artifactId>jackson-datatype-jsr310</artifactId>
-            <version>2.12.3</version>
-        </dependency>
+
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
             <version>1.18.36</version>
             <scope>provided</scope>
         </dependency>
+
+        <!-- Dependências de Teste -->
         <dependency>
-            <groupId>io.github.openfeign</groupId>
-            <artifactId>feign-core</artifactId>
-            <version>11.8</version>
-        </dependency>
-        <dependency>
-            <groupId>io.github.openfeign</groupId>
-            <artifactId>feign-jackson</artifactId>
-            <version>11.8</version>
-        </dependency>
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter</artifactId>
-            <version>5.7.0</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-core</artifactId>
-            <version>3.11.2</version>
+            <groupId>org.springframework.boot</groupId>
```

Comment:
ao adicionar o spring, você está limitando o uso do sdk apenas a esse framework, não? se alguém quiser usar ele com quarkus, micronaut, rxjava ele vai ter dificuldades, entendo que a ideia da criação de um sdk ainda mais para uma api simples, seria facilitar a vida da pessoa que vai utilizar ele.

Reply:
A ideia de adicionar uma versão do SDK com Spring Boot é justamente facilitar a integração com aplicações que já utilizam esse framework. A grande vantagem é que, ao consumir o SDK em projetos com Spring Boot, o próprio framework pode gerenciar as classes e serviços do SDK automaticamente, sem a necessidade de configuração manual. Isso otimiza o processo de desenvolvimento, reduz a duplicação de código e facilita a manutenção, especialmente para regras de negócio e serviços já definidos no SDK. Essa abordagem também facilita a transição para Gateways de Pagamento que utilizam Spring Boot, tornando o processo mais ágil.

Uma versão desacoplada do SDK (sem dependências do Spring) já está em desenvolvimento e garantirá maior flexibilidade, permitindo que o SDK seja facilmente integrado a diferentes ambientes, enquanto mantém a facilidade de uso para quem utiliza Spring Boot.

Vejo um grande potencial de crescimento para o projeto AbacatePay, e por isso, estou criando uma base sólida agora, preparando o SDK para quando for necessário integrá-lo com SpringBoot futuramente.